experiments/ClimaCore/sea_breeze/run.jl

```diff
@@ -111,18 +115,35 @@ lnd_p = (cpl_parameters, F_sfc = lnd_F_sfc)
 land = LandSimulation(lnd_Y, t_start, cpl_Δt / lnd_nsteps, t_end, SSPRK33(), lnd_p, saveat)
 
 # coupled simulation
-struct AOLCoupledSimulation{FT, A <: AtmosSimulation, O <: OceanSimulation, L <: LandSimulation} <:
-       ClimaCoupler.AbstractCoupledSimulation
+struct AOLCoupledSimulation{
+    FT,
+    A <: AtmosSimulation,
+    O <: OceanSimulation,
+    L <: LandSimulation,
+    C <: ClimaCoupler.CouplerState,
+} <: ClimaCoupler.AbstractCoupledSimulation
     # Atmosphere Simulation
     atmos::A
     # Ocean Simulation
     ocean::O
     # Land Simulation
     land::L
+    # Coupler storage
+    coupler::C
     # The coupled time step size
     Δt::FT
 end
-sim = AOLCoupledSimulation(atmos, ocean, land, cpl_Δt)
+
+# init coupler fields and maps
+coupler = CouplerState()
+coupler_add_field!(coupler, :T_sfc_ocean, ocean.integrator.u.T_sfc; write_sim = ocean)
```

Comment:
Another way of asserting that the correct models write into the right vars could be via multiple dispatch, which would allow specifying the same name to one variable in different domains (e.g. `T_sfc(::atm_sim)`, `T_sfc(::ocn_sim)` etc). We can discuss this during the interface 🍐 next week.

Reply:
I'm worried that that would lead to a large number of methods for these shared variables. I think it would be better to have fewer methods that can give us flexibility. Let's discuss further.